src/lib/cimbar_js/cimbar_js.cpp

```diff
@@ -117,17 +127,51 @@ int cimbare_encode(const unsigned char* buffer, unsigned size, const char* filen
 		return -5;
 	}
 
-	Encoder enc;
 	if (encode_id < 0)
-		enc.set_encode_id(++_encodeId); // increment _encodeId every time we change files
+		++_encodeId; // increment _encodeId every time we change files
 	else
-		enc.set_encode_id(static_cast<uint8_t>(encode_id));
+		_encodeId = encode_id;
+
+	_comp = std::make_unique<cimbar::zstd_compressor<std::stringstream>>();
+	if (!_comp)
+		return -1;
+
+	_comp->set_compression_level(_compressionLevel);
+
+	if (fnsize > 0 and filename != nullptr)
+		_comp->write_header(filename, fnsize);
+	return 0;
+}
+
+int cimbare_encode_bufsize()
+{
+	return cimbar::zstd_compressor<std::stringstream>::CHUNK_SIZE;
+}
 
-	cimbar::byte_istream bis(reinterpret_cast<const char*>(buffer), size);
-	_fes = enc.create_fountain_encoder(bis, std::string_view(filename, fnsize), _compressionLevel);
+int cimbare_encode(const unsigned char* buffer, unsigned size)
+{
+	if (!_comp)
+		return -1;
 
+	if (size > 0)
+	{
+		if (!_comp->write(reinterpret_cast<const char*>(buffer), size))
+			return -2;
+	}
+	if (size == cimbare_encode_bufsize())
+		return 1; // more to do
+
+	// otherwise, we're ready
+	unsigned fountainChunkSize = cimbar::Config::fountain_chunk_size();
+	size_t compressedSize = _comp->size();
+	if (compressedSize < fountainChunkSize)
+		_comp->pad(fountainChunkSize - compressedSize + 1);
+
+	// create the encoder stream
+	_fes = fountain_encoder_stream::create(*_comp, fountainChunkSize, _encodeId);
+	_comp.reset();
```

Comment:
After we return 0 from `cimbare_encode`, the next call to it will return `-1` (as if we never called `cimbare_init_encode`)